builder/docker/builder.go

```diff
@@ -120,46 +127,90 @@ func imageBuild(client provision.BuilderDockerClient, app provision.App, imageID
 		ProcfileRaw: procfileBuf.String(),
 		ImageID:     imageID,
 		Out:         evt,
-		CustomData:  customData,
+		CustomData:  tsuruYamlToCustomData(yaml),
 	})
 	if err != nil {
 		return "", err
 	}
 	return newImage, nil
 }
 
-func loadTsuruYaml(client provision.BuilderDockerClient, app provision.App, imageID string, evt *event.Event) (map[string]interface{}, error) {
+func loadTsuruYaml(client provision.BuilderDockerClient, app provision.App, imageID string, evt *event.Event) (*provision.TsuruYamlData, string, error) {
 	path := defaultArchivePath + "/current"
 	cmd := fmt.Sprintf("(cat %s/tsuru.yml || cat %s/tsuru.yaml || cat %s/app.yml || cat %s/app.yaml || true) 2>/dev/null", path, path, path, path)
 	var buf bytes.Buffer
-	err := runCommandInContainer(client, evt, imageID, cmd, app, &buf, nil)
+	containerID, err := runCommandInContainer(client, evt, imageID, cmd, app, &buf, nil)
```

Comment:
What do you think about moving the `defer RemoveContainer(containerID)` call to this function instead of returning the ID and making the parent remove it?

Reply:
The problem is that I call `runCommandInContainer` also in `runBuildHooks` function, and I can't remove it there, because it's commited

Reply:
Oh right, you can only remove it after it's been pushed. 👍